R/zzz.R

```diff
@@ -1,112 +1,45 @@
-# An internal variable that is set ot the name of the native library depending on its use of the Mono or MS.NET CLR
-nativePkgName <- ""
-
-# An internal variable to buffer startup messages
-startupMsg <- ""
-
-#' rClr .onLoad
-#'
-#' Function called when loading the rClr package with 'library'.
-#'
-#' The function looks by default for the rClr native library for the Mono runtime.
-#' If the platform is Linux, this is the only option. If the platform is Windows, using the
-#' Microsoft .NET runtime is an option. If the rClr native library for MS.NET is detected,
-#' the Microsoft .NET runtime is loaded in preference to Mono.
-#'
-#' @param libname the path to the library from which the package is loaded
-#' @param pkgname the name of the package.
-#' @rdname dotOnLoad
-#' @name dotOnLoad
-.onLoad <- function(libname = "~/R", pkgname = "rClr") {
-  rclr_env <- Sys.getenv("RCLR")
-  ext <- .Platform$dynlib.ext
-  nativeLibsNames <- paste("rClrMs", ext, sep = "")
-  msDll <- nativeLibsNames
-  getFnameNoExt <- function(x) {
-    strsplit(x, "\\.")[[1]][1]
-  }
-  rClrPkgDir <- file.path(libname, pkgname)
-  archLibPath <- file.path(rClrPkgDir, "libs")
-  srcPkgLibPath <- NULL
-  if (!file.exists(archLibPath)) {
-    # It may be because this is loaded through the 'document' and 'load_all' functions from devtools,
-    # in which case libname is something like "f:/codeplex"
-    # try to cater for load_all behavior.
-    if ("rclr" %in% tolower(list.files(libname))) {
-      libname <- file.path(rClrPkgDir, "inst")
-      archLibPath <- file.path(rClrPkgDir, "inst/libs")
-      srcPkgLibPath <- archLibPath
-      if (!file.exists(archLibPath)) {
-        stop(paste("Looked like rClr source code directory, but directory not found:", archLibPath))
-      }
-    } else {
-      stop(paste("Trying to work around devtools, but could not find a folder with lowercase name 'rclr' under ", archLibPath))
-    }
-  }
-  dlls <- list.files(archLibPath, pattern = ext)
+# nocov start
+.onLoad <- function(...) {
+  # Check for C++ distrib availability
   if (Sys.info()[["sysname"]] == "Windows") {
-      msvcrFileName <- "msvcp140.dll"
-      if (Sys.which(msvcrFileName) == "") {
-        stop(paste(msvcrFileName, "was not found on this Windows system.",
-          "You are probably missing the Visual C++ Redistributable for Visual Studio 2019.",
-          "Go to https://visualstudio.microsoft.com/downloads/ and dowload 'Microsoft Visual C++ Redistributable for Visual Studio 2019'",
-          sep = "\n"
-        ))
-      }
-      if (!(msDll %in% dlls)) {
-        stop(paste("rClr library .NET framework not found - looked under", archLibPath, "but not found in", paste(dlls, collapse = ",")))
-      }
-      appendStartupMsg("Loading the dynamic library for Microsoft .NET runtime...")
-      chname <- getFnameNoExt(msDll)
-      loadAndInit(chname, pkgname, libname, srcPkgLibPath)
+    if (Sys.which(rNETenv$msvcrFileName) == "") {
```

Comment:
Should we add a similar check for linux? Well once migrated to .NET7

Reply:
In Linux we should check for the glibc version. E.g.

```
> system("ldd --version | grep ldd", intern = TRUE)
[1] "ldd (Ubuntu GLIBC 2.27-3ubuntu1.6) 2.27"
>
```

Once we know which glibc version is required for the package: the installed version must be greater or equal.
E.g.  in the example above, if we require glibc 2.26 or 2.27 - all good, if we require glibc 2.28 - problem.